app/api/discount_codes.py

```diff
@@ -115,9 +154,48 @@ class DiscountCodeDetail(ResourceDetail):
     Discount Code detail by id or code.
     """
 
+    def decide_schema(self, json_data):
+        """To decide discount code schema based on posted data.
+        :param json_data:
+        :return:"""
+
+        used_for = json_data['data']['attributes'].get('used-for')
+        if used_for == 'ticket':
+            self.schema = DiscountCodeSchemaTicket
+        elif used_for == 'event':
+            self.schema = DiscountCodeSchemaEvent
```

Comment:
what if `used_for` is not sent in the PATCH request ? Use the `id` sent in the request and fetch the discount code from the db and use that. Also ensure that the user cannot change the value of `used_for` in a PATCH request.